Sources/Target/Darwin/Thread.cpp

```diff
@@ -33,7 +33,50 @@ namespace Darwin {
 using Host::Darwin::LibProc;
 
 Thread::Thread(ds2::Target::Process *process, ThreadId tid)
-    : super(process, tid), _lastSyscallNumber(-1) {}
+    : super(process, tid), _lastSyscallNumber(-1) {
+  //
+  // Initially the thread is stopped.
+  //
+}
+
+ErrorCode Thread::suspend() {
+  ErrorCode error = kSuccess;
+  if (_state == kRunning) {
+    error = process()->mach().suspend(ProcessThreadId(process()->pid(), tid()));
+    if (error != kSuccess)
+      return error;
+  }
+
+  if (_state == kTerminated) {
+    error = kErrorProcessNotFound;
+  }
+
+  return error;
+}
+
+ErrorCode Thread::resume(int signal, Address const &address) {
+  if (_state == kTerminated) {
+    return kErrorProcessNotFound;
+  }
+
+  if (_state == kInvalid || _state == kRunning) {
+    return kErrorInvalidArgument;
+  }
+
+  signal = 0; // _stopInfo.signal;
```

Comment:
Why do you set signal to 0? You should add a comment to explain this, but I suppose it's because we can't send signals to processes on the Darwin implementation yet. If that's the case, you should just `DS2ASSERT(signal != 0)` at the beginning and mark it as TODO.